examples/multi-agent-debate-system/main.py

```diff
@@ -0,0 +1,299 @@
+    con_ai_agent.dialog.clear()
+    logger.info("AI and Healthcare Pro and Con Agents created")
+
+    # AI and Intellectual Property Pro and Con Agents
+    pro_ip_agent = create_pro_ip_agent(base_llm_config)
+    pro_ip_agent.clear_history(0)
+    pro_ip_agent.dialog.clear()
+    con_ip_agent = create_con_ip_agent(base_llm_config)
+    con_ip_agent.clear_history(0)
+    con_ip_agent.dialog.clear()
+    logger.info("AI and IP Pro and Con Agents created")
+
+    # AI and BIAS Pro and Con Agents
+    pro_bias_agent = create_pro_bias_agent(base_llm_config)
+    pro_bias_agent.clear_history(0)
+    pro_bias_agent.dialog.clear()
+    con_bias_agent = create_con_bias_agent(base_llm_config)
+    con_bias_agent.clear_history(0)
+    con_bias_agent.dialog.clear()
+    logger.info("AI and Bias Pro and Con Agents created")
+
+    # AI and Education Pro and Con Agents
+    pro_edu_agent = create_pro_edu_agent(base_llm_config)
+    pro_edu_agent.clear_history(0)
+    pro_edu_agent.dialog.clear()
+    con_edu_agent = create_con_edu_agent(base_llm_config)
+    con_edu_agent.clear_history(0)
+    con_edu_agent.dialog.clear()
+    logger.info("AI and Education Pro and Con Agents created")
+
+    # Feedback Agent
+    feedback_agent = create_feedback_agent(base_llm_config)
+    feedback_agent.clear_history(0)
+    feedback_agent.dialog.clear()
+    logger.info("Feedback Agent created")
+
+    logger.info("All Agents created successfully.")
+
+
+
+    # Create tasks for each agent using functions from tasks.py
+    tasks = {
+        'pro_ai_task': create_pro_ai_task(pro_ai_agent),
+        'con_ai_task': create_con_ai_task(con_ai_agent),
+        'pro_ip_task': create_pro_ip_task(pro_ip_agent),
+        'con_ip_task': create_con_ip_task(con_ip_agent),
+        'pro_bias_task': create_pro_bias_task(pro_bias_agent),
+        'con_bias_task': create_con_bias_task(con_bias_agent),
+        'pro_edu_task': create_pro_edu_task(pro_edu_agent),
+        'con_edu_task': create_con_edu_task(con_edu_agent),
+        'feedback_task': create_feedback_task(feedback_agent),
+    }
+
+    logger.info("Tasks created successfully.")
+    return tasks
+
+
+# === RUN THE DEBATE LOOP ===
+def run_debate(tasks):
+    """
+    Runs the debate interaction logic.
+
+    Args:
+        tasks (dict): A dictionary of available debate tasks.
+    """
+
+    logger.info("Starting debate session.")
+
+    # Prompt for LLM delegate setting
+    llm_delegate = get_llm_delegate_setting()
+    logger.info("LLM Delegate setting: %s", llm_delegate)
+
+    # === Select Debate Topic ===
+    debate_topic = Prompt.ask(
+        "Select a debate topic:\n1. AI in Healthcare\n2. AI and Intellectual Property Rights\n"
+        "3. AI and Societal Biases\n4. AI as an Educator",
+        choices=["1", "2", "3", "4"],
+        default="1",
+    )
+    logger.info("Debate topic selected: %s", debate_topic)
+
+    # Selecting the topic and the appropriate Pro or Con agent
+    if debate_topic == "1":
```

Comment:
all these cases can be handled without having to have multiple if/then statements, using a bit more extra code.